pkg/controller/iam/user/controller.go

```diff
@@ -162,6 +175,38 @@ func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.Ex
 		UserName: aws.String(meta.GetExternalName(cr)),
 	})
 
+	if err != nil {
+		return managed.ExternalUpdate{}, awsclient.Wrap(err, errUpdate)
+	}
+
+	observed, err := e.client.GetUser(ctx, &awsiam.GetUserInput{
+		UserName: aws.String(meta.GetExternalName(cr)),
+	})
+
+	if err != nil {
+		return managed.ExternalUpdate{}, awsclient.Wrap(err, errGet)
+	}
+
+	add, remove, _ := iam.DiffIAMTagsWithUpdates(cr.Spec.ForProvider.Tags, observed.User.Tags)
+
+	if len(add) > 0 {
```

Comment:
Is there a possibility that both `remove` and `add` have the same entry since it needs to be updated so we end up first updating it here and then remove it in the next `if` block?

Reply:
not with `DiffIAMTagsWithUpdates`. 

The use case you're mentioning is covered by [this test](https://github.com/crossplane/provider-aws/blob/master/pkg/clients/iam/tags_test.go#L163), and more generically [here](https://github.com/cebernardi/provider-aws/blob/add-default-tags-iam-user/pkg/controller/iam/user/controller_test.go#L452).